### DISCUSSION TITLE
About -user

### DISCUSSION THREAD

**Discussion by steakhutzeee** [prodrigestivill/docker-postgres-backup-local#100] [Q&A] [answered]
Hi, would like to ask clarification about:

> For security reasons it is recommended to run it as user postgres:postgres.
> 
> In case of running as postgres user, the system administrator must initialize the permission of the destination folder as follows:

```
# for default images (debian)
mkdir -p /var/opt/pgbackups && chown -R 999:999 /var/opt/pgbackups
# for alpine images
mkdir -p /var/opt/pgbackups && chown -R 70:70 /var/opt/pgbackups
```

Why i need to run with postgres user?

Thanks in advance!

**Comment by prodrigestivill** [comment, marked as answer]
It has been documented as requested by #29.
But in general is a good practice to not run programs as root.

More information can be found in [official docker documentation](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#user).

**Comment by steakhutzeee** [comment]
> 
> 
> It has been documented as requested by #29.
> But in general is a good practice to not run programs as root.
> 
> More information can be found in [official docker documentation](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#user).

Thanks! 

Well, i suppose the user running "docker-postgres-backup-local" should be the same running the db, otherwise it could not access the db to back it up. Correct?

Per my understanding PostgreSQL runs as user "postgres" by default, so it's ok to run also your container as "postgres" user.
